### PR TITLE
Create-able Select components

### LINKED ISSUE #1710: Create-able MultiSelect
### MultiSelect with Create-able options.

Currently MultiSelect can select from a given list. 

Would like an option where given a prop(say allowCreate), it allows to create a new Tag from a user which doesn't exist in the predefined list.

### PR DESCRIPTION
**Pull request by shuyangli**
#### Fixes #1710 

- [x] Includes tests
- [x] Update documentation

#### Changes proposed in this pull request:
- Select components, via `IListItemsProps`, now takes two additional props:
  * `createItemFromQuery?: (query: string) => T` which converts a query to a new item; if this prop is present, the select component is considered "create-able";
  * `createItemRenderer?: (query: string, handleClick: React.MouseEventHandler<HTMLElement>) => JSX.Element | undefined` which could be used to render a custom `MenuItem` at the end of the list, that represents "create a new element from query".
- If the component is create-able, when a query returns no results, pressing "enter" key will create a new item with the current query;

### CODE REVIEW

File: packages/docs-app/src/examples/select-examples/films.tsx [hunk 1]
```diff
@@ -182,3 +182,11 @@ export const filmSelectProps = {
     itemRenderer: renderFilm,
     items: TOP_100_FILMS,
 };
+
+export function createFilm(title: string): IFilm {
+    return {
+        rank: 100 + Math.floor(Math.random() * 100 + 1),
```
Comment (giladgray) [inline packages/docs-app/src/examples/select-examples/films.tsx]: 😂

File: packages/select/src/components/select/select-component.md [hunk 2]
```diff
@@ -82,6 +82,12 @@ This controlled usage allows you to implement all sorts of advanced behavior on
 top of the basic `Select` interactions, such as windowed filtering for large
 data sets.
 
+@## Creating new items
+
+Use the `createItemFromQuery` prop to allow `Select<T>` to create new items. `createItemFromQuery` specifies how to turn a
+user-entered query string into an item of type `<T>` that `Select` understands. You should also provide the `createItemRenderer`
+prop to render a custom `MenuItem` to indicate that users can create new items.
```
Comment (giladgray) [inline packages/select/src/components/select/select-component.md]: let's add a simple code sample here that demonstrates using both props.

File: packages/select/test/selectComponentSuite.tsx [hunk 3]
```diff
@@ -119,4 +119,34 @@ export function selectComponentSuite<P extends IListItemsProps<IFilm>, S>(
...
+        const testCreateProps = {
+            ...testProps,
+            createItemFromQuery: sinon.spy(),
+            createItemRenderer: () => <textarea />,
+        };
+
+        it("renders create item if filtering returns empty list", () => {
+            const wrapper = render({
+                ...testCreateProps,
+                items: [],
```
Comment (giladgray) [inline packages/select/test/selectComponentSuite.tsx]: use the real item list so it's actually filtering instead of always being an empty list

**Comment by shuyangli** [comment]
Closing in favor of #3306

**Pull request closed without merging**